tests/e2e/coverage/Dockerfile

```diff
@@ -0,0 +1,37 @@
+# Build the manager binary
+FROM golang:1.23 as builder
+ARG TARGETOS
+ARG TARGETARCH
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY cmd/main.go cmd/main.go
+COPY pkg/ pkg/
+COPY internal/ internal/
+
+# Add a go build cache. The persistent cache helps speed up build steps,
+# especially steps that involve installing packages using a package manager.
+ENV GOCACHE=/root/.cache/go-build
+# Build
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
```

Comment:
I know this was copied from the other Dockerfile, but do you mind fixing the phrasing to "the GOARCH doesn't have a default value"?